ros2_batch_job/__main__.py

```diff
@@ -86,6 +86,9 @@ def get_args(sysargv=None):
         '--repo-file-url', required=True,
         help="url of the ros2.repos file to fetch and use for the basis of the batch job")
     parser.add_argument(
+        '--supplemental-repo-file-url', default=None,
+        help="url of a .repos file to fetch and merge with the ros2.repos file")
```

Comment:
It could. But I would be inclined not to until we have a scenario where it would be strongly preferred over composing an omnibus supplemental or primary repos file and I'd be inclined to implement it directly in vcstool, passing the list of repos files as multiple `--input`s directly through and merging them before actually fetching repositories.

The UX cost of dumping additional urls into build parameters is high, using a string parameter would make it hard to scan the list of repos files and I don't see a list subtype of [ParameterDefinition](http://javadoc.jenkins.io/hudson/model/ParameterDefinition.html) that would give a reasonable and flexible experience. If we did it, I'd probably change the supplemental repos parameter to a text type and parse one url per line.

On top of the UX problem, which I am more concerned about, the cost of re-running vcstool isn't that high at the moment, the first run takes 13-15 seconds and running twice takes between 17-22 seconds but compounded over 3-7 files or something pathological like one repo per file and you'd lose most of the advantage of vcs running in parallel.

Both of these have short and long term solutions, and I do want to factor the code to make it an easy change later, but I don't feel a desire to encourage it by making the parameters work that way.